app/src/main/java/de/jakobclass/transittracker/MapActivity.kt

```diff
@@ -1,43 +1,102 @@
 package de.jakobclass.transittracker
 
+import android.Manifest
+import android.app.AlertDialog
+import android.content.pm.PackageManager
+import android.location.Location
 import android.support.v4.app.FragmentActivity
 import android.os.Bundle
+import android.support.v4.app.ActivityCompat
+import android.support.v4.content.ContextCompat
+import com.google.android.gms.common.api.GoogleApiClient
+import com.google.android.gms.common.api.GoogleApiClient.ConnectionCallbacks
+import com.google.android.gms.location.LocationServices
 
 import com.google.android.gms.maps.CameraUpdateFactory
 import com.google.android.gms.maps.GoogleMap
 import com.google.android.gms.maps.OnMapReadyCallback
 import com.google.android.gms.maps.SupportMapFragment
 import com.google.android.gms.maps.model.LatLng
-import com.google.android.gms.maps.model.MarkerOptions
 
-class MapActivity : FragmentActivity(), OnMapReadyCallback {
+class MapActivity : FragmentActivity(), OnMapReadyCallback, ConnectionCallbacks {
 
     private var map: GoogleMap? = null
+    private var googleApiClient: GoogleApiClient? = null
+
+    private val REQUEST_CODE_ACCESS_FINE_LOCATION = 1
+    private val REQUIRED_LOCATION_PREMISSION = Manifest.permission.ACCESS_FINE_LOCATION
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_map)
-        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
         val mapFragment = supportFragmentManager.findFragmentById(R.id.map) as SupportMapFragment
         mapFragment.getMapAsync(this)
     }
 
+    override fun onStop() {
```

Comment:
I skipped that because I only need the location once (for setting the initial position). So I only connect in line 83. But yeah, when I start tracking the user's location I'd have to reconnect in `onStart`